src/communication/kernels/reduce.cpp

```diff
@@ -8,8 +8,8 @@
 // SPDX-License-Identifier: BSD-3-Clause
 //
 
-#ifdef DLAF_WITH_CUDA_RDMA
-#warning "Reduce is not using CUDA_RDMA."
+#ifdef DLAF_WITH_MPI_GPU_SUPPORT
+#warning "Reduce is not using MPI_GPU_SUPPORT."
```

Comment:
@albestro do you think we still need this warning? It's not so good with `-Werror`. If we still think it's possible to make reduce support GPU memory directly we should perhaps track it in an issue instead?

Reply:
I'm ok with removing it.

It was just a placeholder to make the user aware that even if MPI with GPU support is enabled and it works, actually `reduce` is not using it.

In case we decide to track that with an issue, it won't be more hidden than it is now. I'm not even sure it is something we want to track.

Reply:
Not sure why the reduce was not implemented.
The only requirement is to have contiguous memory (but the requirement is present in the CPU version as well)

Reply:
So I would go for tracking it with an issue so we can eventually implement it. 😄

Reply:
https://github.com/eth-cscs/DLA-Future/issues/1094

Reply:
Thanks @albestro!